base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -281,8 +281,10 @@ where
             futures::select! {
                 //Incoming request
                 request_context = request_stream.select_next_some() => {
-                    trace!(target: LOG_TARGET, "Handling Service API Request");
+                    let start = Instant::now();
```

Comment:
I am not a fan of adding all this code to log timings that could be found via log analysis, if you really think its worth leaving in then at least reduce it to a single timing from the top of the loop to the bottom rather than so many individual timing blocks in an already cumbersome select! statement.

Reply:
Yeah, we specifically needed to look at the timings within the `futures::select! {` here.